controllers/discovery_stroage.go

```diff
@@ -0,0 +1,144 @@
+package controllers
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/go-logr/logr"
+	"github.com/open-policy-agent/gatekeeper/v3/apis/config/v1alpha1"
+	"github.com/pkg/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/kubernetes"
+)
+
+var ErrNotFoundDiscovery = errors.New("There is no matched apiGroup, version or kind")
```

Comment:
This would be a lint error in our other repos. I'm not sure why there isn't linting enabled to catch this.
```suggestion
var ErrNotFoundDiscovery = errors.New("there is no matched apiGroup, version or kind")
```